dace/optimization/on_the_fly_map_fusion_tuner.py

```diff
@@ -93,7 +93,8 @@ def evaluate(self, config, cutout, measurements: int, **kwargs) -> float:
         maps_ = list(map(candidate.start_state.node, map_ids))
         subgraph = helpers.subgraph_from_maps(sdfg=candidate, graph=candidate.start_state, map_entries=maps_)
 
-        map_fusion = sg.SubgraphOTFFusion(subgraph, candidate.sdfg_id, candidate.node_id(candidate.start_state))
+        map_fusion = sg.SubgraphOTFFusion()
+        map_fusion.setup_match(subgraph, candidate.sdfg_id, candidate.node_id(candidate.start_state))
```

Comment:
Are these warnings from codecov correct? If you cannot see them, it basically says "Added lines #L96 - L97 were not covered by tests." Since there seem quite a few of those in this file, should we ask @lukastruemper to write some suitable tests?

Reply:
Yes, will do! Re-writing parts of the tuner code currently.

I am actually thinking about even re-writing the fusion transformations. To me, it seems like we have multiple fusion transformations which are not really disjoint.